tests/test_reductions_further.py

```diff
@@ -0,0 +1,395 @@
+                                  dim_name=dim_name_2)
+group_unique_2 = np.unique(dim_2.month)
+group_name_2 = "month"
+values_2 = np.array(
+      [[[np.nan,  0., 57.],
+        [57., 57., 57.],
+        [ 0.,  0.,  57.]],
+
+       [[ 0.,  0., 57.],
+        [ 0.,  0.,  0.],
+        [ 0., 57.,  0.]],
+
+       [[ 0., 57.,  0.],
+        [57.,  0.,  0.],
+        [ 0., 57., 57.]],
+
+       [[57.,  0.,  0.],
+        [57.,  0.,  0.],
+        [57., 57., 57.]]])
+
+
+data_2 = create_data_array(values = values_2, 
+                           dim = dim_2, 
+                           dim_name=dim_name_2)
+# also create a dataset
+dataset = data_2.to_dataset(dim=None, name="test", promote_attrs=False)
+
+weighted_mean_2 = np.array(
+      [[[57.,  0., 29.],
+        [57., 29., 29.],
+        [28., 28., 57.]],
+
+       [[ 0.,  0., 57.],
+        [ 0.,  0.,  0.],
+        [ 0., 57.,  0.]],
+
+       [[ 0., 57.,  0.],
+        [57.,  0.,  0.],
+        [ 0., 57., 57.]]])
+weighted_mean_2 = create_mean_array(mean = weighted_mean_2,
+                                        group_unique=group_unique_2,
+                                        group_name = group_name_2)
+unweighted_mean_2 = np.array(
+      [[[57,  0., 28.5,],
+        [57, 28.5, 28.5,],
+        [28.5, 28.5,  57]],
+
+       [[ 0.,  0., 57.],
+        [ 0.,  0.,  0.],
+        [ 0., 57.,  0.]],
+
+       [[ 0., 57.,  0.],
+        [57.,  0.,  0.],
+        [ 0., 57., 57.]]])
+unweighted_mean_2 = create_mean_array(mean = unweighted_mean_2,
+                                        group_unique=group_unique_2,
+                                        group_name = group_name_2)
+
+
+# monthly anomalies for the data. 
+# nans will be included!
+weighted_anomalies_2 = np.    array(
+        [[[np.nan,   0.,  28.],
+            [  0.,  28.,  28.],
+            [-28., -28.,   0.]],
+
+           [[  0.,   0.,   0.],
+            [  0.,   0.,   0.],
+            [  0.,   0.,   0.]],
+
+           [[  0.,   0.,   0.],
+            [  0.,   0.,   0.],
+            [  0.,   0.,   0.]],
+
+           [[  0.,   0., -29.],
+            [  0., -29., -29.],
+            [ 29.,  29.,   0.]]])
+weighted_anomalies_2 = create_data_array(values = weighted_anomalies_2, 
+                                  dim = dim_2, 
+                                  dim_name=dim_name_2)
+
+unweighted_anomalies_2 = np.array(
+     [[[np.nan,   0. ,  28.5],
+        [  0. ,  28.5,  28.5],
+        [-28.5, -28.5,   0. ]],
+
+       [[  0. ,   0. ,   0. ],
+        [  0. ,   0. ,   0. ],
+        [  0. ,   0. ,   0. ]],
+
+       [[  0. ,   0. ,   0. ],
+        [  0. ,   0. ,   0. ],
+        [  0. ,   0. ,   0. ]],
+
+       [[  0. ,   0. , -28.5],
+        [  0. , -28.5, -28.5],
+        [ 28.5,  28.5,   0. ]]])
+unweighted_anomalies_2 = create_data_array(values = unweighted_anomalies_2, 
+                                  dim = dim_2, 
+                                  dim_name=dim_name_2)
```

Comment:
Same applies here as in the other test module: The pattern of declaring data first, and then using very concise tests, is very difficult to understand. I'd rather go for explicit code and repetition rather than elegance here.